prusti-tests/tests/verify_overflow/pass/competitve_programming/dp/SumOfDifferentPrimes_1213.rs

```diff
@@ -0,0 +1,237 @@
+    }
+
+    #[trusted]
+    #[ensures (result.len() == 0)]
+    pub fn new() -> Self {
+        VecWrapperI32 { v: Vec::new() }
+    }
+
+    #[trusted]
+    #[pure]
+    #[requires (0 <= index && index < self.len())]
+    pub fn lookup(&self, index: isize) -> isize {
+        self.v[index as usize]
+    }
+
+    #[trusted]
+    #[ensures (self.len() == old(self.len()) + 1)]
+    #[ensures (self.lookup(old(self.len())) == value)]
+    #[ensures (forall(|i: isize| (0 <= i && i < old(self.len())) ==>
+                    self.lookup(i) == old(self.lookup(i))))]
+    pub fn push(&mut self, value: isize) {
+        self.v.push(value);
+    }
+}
+
+struct Matrix {
+    _ghost_y_size: usize,
+    _ghost_x_size: usize,
+    vec: Vec<Vec<isize>>,
+}
+
+impl Matrix {
+    #[trusted]
+    #[requires(0 < y_size)]
+    #[requires(0 < x_size)]
+    #[ensures(result.y_size() == y_size)]
+    #[ensures(result.x_size() == x_size)]
+    #[ensures(forall(|y: isize, x: isize|
+                (0 <= x && x < result.x_size() && 0 <= y && y < result.y_size()) ==>
+                result.lookup(y, x) == 0))]
+    fn new(y_size: isize, x_size: isize) -> Self {
+        Self {
+            _ghost_y_size: y_size as usize,
+            _ghost_x_size: x_size as usize,
+            vec: vec![vec![0; y_size as usize]; x_size as usize],
+        }
+    }
+
+    #[pure]
+    #[trusted]
+    #[ensures(0 < result)]
+    fn x_size(&self) -> isize {
+        self._ghost_x_size as isize
+    }
+
+    #[pure]
+    #[trusted]
+    #[ensures(0 < result)]
+    fn y_size(&self) -> isize {
+        self._ghost_y_size as isize
+    }
+
+    #[trusted]
+    #[requires(0 <= y && y < self.y_size())]
+    #[requires(0 <= x && x < self.x_size())]
+    #[ensures(self.y_size() == old(self.y_size()))]
+    #[ensures(self.x_size() == old(self.x_size()))]
+    #[ensures(forall(|i: isize, j: isize|
+        (i >= 0 && i < y && j >= 0 && j < self.x_size()) ==> (self.lookup(i, j) == old(self.lookup(i, j)))))]
+    #[ensures(self.lookup(y, x) == value)]
+    #[ensures(forall(|i: isize, j: isize|
+        (0 <= i && i < self.y_size() &&
+         0 <= j && j < self.x_size() && !(j == x && i == y)) ==>
+        self.lookup(i, j) == old(self.lookup(i, j))))]
+    fn set(&mut self, y: isize, x: isize, value: isize) -> () {
+        self.vec[y as usize][x as usize] = value
+    }
+
+    #[pure]
+    #[trusted]
+    #[requires(0 <= y && y < self.y_size())]
+    #[requires(0 <= x && x < self.x_size())]
+    fn lookup(&self, y: isize, x: isize) -> isize {
+        self.vec[y as usize][x as usize]
+    }
+}
+
+// Recursive solution
+
+#[pure]
+#[requires(n <= 1120)]
+#[requires(k <= 14)]
+#[requires(primes.len() > 0)]
+#[requires(forall(|k: isize| (k >= 0 && k < primes.len()) ==> (primes.lookup(k) >= 2)))]
+//#[ensures(k > 0 && n >= 0 ==> result  == sum_of_different_primes_rec_helper(primes, n, k, primes.len() - 1))]
+fn sum_of_different_primes_rec(primes: &VecWrapperI32, n: isize, k: isize) -> isize {
```

Comment:
@vakaras Here is the function that reports the error wrongly with checking overflow.